File: two-sum-1/two_sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> map;
        for(int i=0; i<nums.size(); i++) {
            int n = target - nums[i];
            if(map.count(n)>0){
                return {i, map[n]};
            }
            map[nums[i]]=i;
        }
        return {};
    }
};

File: two-sum-1/two_sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:

        _map = {}

        for i,num in enumerate(nums):
            n = target - num
            if n in _map:
                return [i, _map[n]]
            _map[num] = i
        